packages/@ionic/cli/src/commands/capacitor/run.ts

```diff
@@ -222,4 +229,39 @@ For Android and iOS, you can setup Remote Debugging on your device with browser
       `To continue, run your project on a device or ${getVirtualDeviceNameForPlatform(platform)} using ${getNativeIDEForPlatform(platform)}!`
     );
   }
+
+  private async runCapacitorHook(project: IProject , inputs: CommandLineInputs, options: CommandLineOptions, e: HookDeps): Promise<void> {
+    const hook = new CapacitorRunHook(e);
+    let serveOptions: ServeOptions | undefined;
+    let buildOptions: any | undefined;
+
+    if (options['livereload']) {
+      const serveRunner = await project.requireServeRunner();
+
+      serveOptions = serveRunner.createOptionsFromCommandLine(inputs, options);
+    } else {
+      const buildRunner = await project.requireBuildRunner();
+
+      buildOptions = buildRunner.createOptionsFromCommandLine(inputs, options);
+    }
+
+    try {
+      await hook.run({
+        name: hook.name,
+        serve: serveOptions,
+        build: buildOptions,
+        capacitor: this.createOptionsFromCommandLine(inputs, options),
+      });
+    } catch (e) {
+      if (e instanceof BaseError) {
+        throw new FatalException(e.message);
+      }
+
+      throw e;
+    }
+  }
+}
+
+class CapacitorRunHook extends Hook {
+  readonly name = 'capacitor:run';
```

Comment:
I wonder if we should `capacitor:run:after` and `capacitor:build:after` instead just so it's clear when they happen?

Reply:
Changed the hook names as suggested :-)